application.py

```diff
@@ -513,6 +513,27 @@ def contest(contest_id):
     return render_template("contest/contest.html", title=title, scoreboard=scoreboard,
                            data=data)
 
+@app.route("/contest/<contest_id>/notify", methods=['GET', 'POST'])
+@admin_required
+def contest_notify(contest_id):
+    if request.method == "GET":
+        return render_template('admin/contestnotify.html')
+
+    subject = request.form.get("subject")
+    if not subject:
+        return "Must provide subject", 400
+    message = request.form.get("message")
+    if not message:
+        return "Must provide message", 400
+
+    data = db.execute("SELECT email FROM contest_users JOIN users on user_id=users.id WHERE contest_users.contest_id=:cid ORDER BY points DESC, lastAC ASC",
```

Comment:
don't think it's necessary to order by points and lastAC, consider getting rid of the ORDER BY part if it's not necessary.